tests/tracer/test_rate_limiter.py

```diff
@@ -150,47 +156,51 @@ def test_rate_limiter_effective_rate_starting_rate(time_window):
+    with mock.patch("ddtrace.internal.rate_limiter.compat.monotonic_ns", return_value=now_ns):
+        assert limiter.is_allowed() is True
+        assert limiter.effective_rate == 1.0
+        assert limiter.current_window_ns == now_ns
+        assert limiter.prev_window_rate is None
 
     # Gap of 0.9999 seconds, same window
     time_ns = now_ns + (0.9999 * time_window)
-    assert limiter.is_allowed(time_ns) is False
-    # DEV: We have rate_limit=1 set
-    assert limiter.effective_rate == 0.5
-    assert limiter.current_window_ns == now_ns
-    assert limiter.prev_window_rate is None
+    with mock.patch("ddtrace.internal.rate_limiter.compat.monotonic_ns", return_value=time_ns):
+        assert limiter.is_allowed() is False
+        # DEV: We have rate_limit=1 set
+        assert limiter.effective_rate == 0.5
+        assert limiter.current_window_ns == now_ns
+        assert limiter.prev_window_rate is None
 
     # Gap of 1.0 seconds, new window
     time_ns = now_ns + time_window
-    assert limiter.is_allowed(time_ns) is True
-    assert limiter.effective_rate == 0.75
-    assert limiter.current_window_ns == (now_ns + time_window)
-    assert limiter.prev_window_rate == 0.5
+    with mock.patch("ddtrace.internal.rate_limiter.compat.monotonic_ns", return_value=time_ns):
+        assert limiter.is_allowed() is True
+        assert limiter.effective_rate == 0.75
+        assert limiter.current_window_ns == (now_ns + time_window)
+        assert limiter.prev_window_rate == 0.5
 
     # Gap of 1.9999 seconds, same window
     time_ns = now_ns + (1.9999 * time_window)
-    assert limiter.is_allowed(time_ns) is False
-    assert limiter.effective_rate == 0.5
-    assert limiter.current_window_ns == (now_ns + time_window)  # Same as old window
-    assert limiter.prev_window_rate == 0.5
+    with mock.patch("ddtrace.internal.rate_limiter.compat.monotonic_ns", return_value=time_ns):
+        assert limiter.is_allowed() is False
+        assert limiter.effective_rate == 0.5
+        assert limiter.current_window_ns == (now_ns + time_window)  # Same as old window
+        assert limiter.prev_window_rate == 0.5
 
     # Large gap of 100 seconds, new window
     time_ns = now_ns + (100.0 * time_window)
-    assert limiter.is_allowed(time_ns) is True
-    assert limiter.effective_rate == 0.75
-    assert limiter.current_window_ns == (now_ns + (100.0 * time_window))
-    assert limiter.prev_window_rate == 0.5
+    with mock.patch("ddtrace.internal.rate_limiter.compat.monotonic_ns", return_value=time_ns):
+        assert limiter.is_allowed() is True
+        assert limiter.effective_rate == 0.75
+        assert limiter.current_window_ns == (now_ns + (100.0 * time_window))
+        assert limiter.prev_window_rate == 0.5
 
 
 def test_rate_limiter_3():
     limiter = RateLimiter(rate_limit=2)
 
-    now_ns = compat.monotonic_ns()
     for i in range(3):
-        decision = limiter.is_allowed(now_ns)
+        decision = limiter.is_allowed()
```

Comment:
we have to mock the time here, otherwise it is a race condition on time window changing.